### PR TITLE
Updated transform-react-display-name for createReactClass addon

### PR DESCRIPTION
**Pull request by bvaughn**
| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          | 
| Major: Breaking Change?  | 
| Minor: New Feature?      | Yes
| Deprecations?            | 
| Spec Compliancy?         | 
| Tests Added/Pass?        | Yes
| Fixed Tickets            | 
| License                  | MIT
| Doc PR                   | 
| Dependency Changes       | 

The React team will be deprecating `React.createClass` soon (see facebook/react/pull/9232). This method will be moving to a new add-on package `create-react-class` (~~not yet released~~ **edit** Released on Friday, April 7). We are in the process of creating a codemod now that will replace:
```js
import React from 'react';

React.createClass({...})
```
with:
```js
import createReactClass from 'create-react-class';

createReactClass({...})
```

This diff updates the `transform-react-display-name` plugin to handle the new naming convention (`createReactClass`) as well as the pre-existing use case (`React.createClass`) in order to remain backwards compatible with existing versions of React. The new check is a naive string comparison and so it's pretty weak- but hopefully this is sufficient given that `createClass` is deprecated going forward.

**Caveat**: It is possible that someone will manually write code using the new add-on that does not conform to the naming convention used by the codemod. This use case is not supported.

cc @acdlite

### CODE REVIEW

**Comment by hzoo** [comment]
@bvaughn yeah the only other relevant issue I remember is https://github.com/babel/babel/issues/4663 which we never merged

**Comment by bvaughn** [comment]
This PR is a bit more narrow-scope than that issue since it's only concerned with `createClass` usage. Functions/classes generally have a `name` attribute that React [will already use](https://github.com/facebook/react/blob/b1b4a2fb252f26fe10d29ba60d85ff89a85ff3ec/src/shared/utils/getComponentName.js#L34) instead of the `displayName` so I'm not sure how pressing the other request is. Maybe there are use cases I'm not thinking about?

File: packages/babel-plugin-transform-react-display-name/README.md [hunk 1]
```diff
@@ -7,15 +7,19 @@
 **In**
 
 ```js
-var foo = React.createClass({});
+var foo = React.createClass({}); // React <= 15
```
Comment (hzoo) [inline packages/babel-plugin-transform-react-display-name/README.md]: dono if we want to modify ` > Add displayName to React.createClass calls` above this too

Comment (existentialism) [inline packages/babel-plugin-transform-react-display-name/README.md]: We should definitely update the docs after the codemod/package are published

**Comment by acdlite** [comment]
Let's use `createReactClass` instead

File: packages/babel-plugin-transform-react-display-name/README.md (hunk 1 above)
Comment (bvaughn) [inline packages/babel-plugin-transform-react-display-name/README.md]: Sure thing fellas. Sorry it took me a bit to circle back here. Got pulled away on some react-native stuff. Wording has been updated.

**Comment by bvaughn** [comment]
Sounds like we have decided to go with `createReactClass` instead of `ReactCreateClass`~ so I've made one last update to this transform. Sorry for the churn!

**Comment by acdlite** [comment]
Let's maybe wait to merge this until 15.5 is released, just in case

**Comment by hzoo** [comment]
FYI this will be targeting 7.0 anyway unless you also do a backport to 6.x

**Comment by hzoo** [comment]
Also adding both of you to a "react" team on babel (read access) so that we can assign people if that's cool (who else should I add)

**Comment by acdlite** [comment]
Dan is already added I think. The rest of the team is sebmarkbage, spicyj, trueadm, and flarnie.

**Comment by hzoo** [comment]
the flow I was thinking of

issues: in an updated version of the issue template maybe can specify a place for a user to auto label an issue.
pr: the babel-bot can read if the files touch affect `packages/babel-plugin-*-react-*` and then @ mention the relevant team. Or just modify mention-bot to do it (easier).

**Comment by timdorr** [comment]
/poke

Looks like I borked a release on react-router 3.x, since we use createClass in that version. Would love to get this out and backported to 6.x. Let me know if I can help!

**Comment by birdy-** [comment]
LGTM

**Comment by hzoo** [comment]
@timdorr yeah feel free to do the backport pr

**Comment by itkach** [comment]
Any chance to have this backported to babel 6?

**Comment by bvaughn** [comment]
>> **@timdorr**: Looks like I borked a release on react-router 3.x, since we use createClass in that version. Would love to get this out and backported to 6.x. Let me know if I can help!

> **@hzoo**: @timdorr yeah feel free to do the backport pr

@itkach Seems like someone would just need to submit a PR

**Pull request merged**